code/modules/mob_spawn/ghost_roles/spider_roles.dm

```diff
@@ -102,11 +102,9 @@
 			to_chat(user, span_warning("\The [src] is not ready to hatch yet!"))
 		return FALSE
 
-/obj/effect/mob_spawn/ghost_role/spider/equip(mob/living/simple_animal/hostile/giant_spider/spawned_spider)
-	if(spawned_spider)
-		spawned_spider.directive = directive
-
-/obj/effect/mob_spawn/ghost_role/spider/special(mob/user)
+/obj/effect/mob_spawn/ghost_role/spider/special(mob/living/simple_animal/hostile/giant_spider/spawned_mob, mob/mob_possessor)
+	if (spawned_mob)
+		spawned_mob.directive = directive
 	. = ..()
```

Comment:
Any specific reason why we set the directive before we call parent?

Reply:
Yes, in parent it assigns the mind and then in the `login` proc for spiders it assigns the antag datum, which prints their objectives into the chat and pops up the antag panel.
We need to give them their objective _before_ that or it doesn't show it to them, which is a bug on live at the moment.
This is pretty fragile but I couldn't think of a way to make it less so.

Reply:
Why can't we just `return ..()` at the end of the proc then?

Reply:
You're right, the egg cleanup is old code but I don't know why I didn't just move all of it above the super call.